src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java

```diff
@@ -26,7 +25,12 @@ public User add(User user) {
     }
 
     @Override
-    public Optional<User> findByEmail(String email) {
-        return userDao.findByEmail(email);
+    public User getById(Long id) {
+        return userDao.getById(id).get();
+    }
+
+    @Override
+    public User findByEmail(String email) {
+        return userDao.findByEmail(email).get();
```

Comment:
I would say that it is better to still return Optional here (and call optional.get() in UserController)

the problem is that, if user enters wrong email, previously the AuthenticationExeption was thrown from AuthService - and now NoSuchElementException (you call `get()` on empty Optional in UserServiceImpl)

Reply:
Ok, ive only changed it according to our qna :)

Reply:
sometimes one may not notice some issues from the first glance)

Reply:
Youre right, ofcourse )